pkg/pillar/cmd/volumemgr/initialvolumestatus.go

```diff
@@ -72,6 +75,35 @@ func gcObjects(ctx *volumemgrContext, dirName string) {
 	log.Tracef("gcObjects(%s) Done", dirName)
 }
 
+// Periodic garbage collection of children datasets in provided zfs dataset
+func gcDatasets(ctx *volumemgrContext, dataset string) {
```

Comment:
We have the same for files. I think, it may be useful for some broken datasets/files we keep

Reply:
@rvs In general we need GC to handle objects which are deleted on the controller while EVE is down/rebooting. 
In that case we have various agents in EVE which scan their state in /persist to e.g., build up some status, and then after EVE has received the initial config from the controller we run a GC function which looks for status without a corresponding config. The details differ between this being e.g., a UUID to appnum or bridgenum (which we persist), or content-trees and volumes created in /persist. But the need is the same.

In this case, if the VolumeConfig was deleted on the controller while EVE was down we do not see a handleDelete, hence need the GC to clean up.
(But maybe we should redo this and have EVE on boot always start by inhaling the last checkpointed config before it connects to the controller because in that case we will see delete callbacks for items which have been deleted. But that wouldn't be part of this PR ;-)